src/plugins/discourse/mirror.test.js

```diff
@@ -110,63 +188,158 @@ class MockFetcher implements Discourse {
   }
 
   _post(id: PostId): Post | null {
-    const postInfo = this._posts.get(id);
-    if (postInfo == null) {
-      return null;
+    return this._posts.get(id) || null;
+  }
+
+  /**
+   * Adds a new Category.
+   * This will create a category definition topic as well.
+   * The definition topic will create an opening post.
+   *
+   * Note: does not have any perception of subforums.
+   */
+  addCategory(topic?: $Shape<TopicOptions>): CreatedCategory {
+    // Have the new Category exist.
+    const categoryId = this._nextCategoryId;
```

Comment:
`const categoryId = this._nextCategoryId++` is sufficient (though `const categoryId = ++this._nextCategoryId` would have the wrong behavior). that said, your way is unambiguously correct and the concise way requires a little more familiarity with the operator, so no change requested.

Reply:
Right, that was my intention here. I'm aware of `var++` and `++var` syntax. But putting them on separate lines makes it a lot easier to see these are two semantic operations and what the intended order is.